src/test/compile-fail/unsized5.rs

```diff
@@ -24,7 +24,7 @@ struct S3 {
     g: [usize]
 }
 struct S4 {
-    f: str, //~ ERROR `core::marker::Sized` is not implemented
+    f: [u8], //~ ERROR `core::marker::Sized` is not implemented
```

Comment:
I think I would try to do this in a way that doesn't worsen error messages much.